keys-component.js

```diff
@@ -59,7 +72,17 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
     self.modals.key = null;
   };
 
-  self.init = function(identity) {
+  self.onKeyGenerate = function(key) {
+    return new Promise(function(resolve) {
```

Comment:
With this test promise it works properly 12/12 with all the other test logging removed.